applications/utils/reservation_creation.py

```diff
@@ -0,0 +1,63 @@
+import datetime
+import logging
+
+from django.db import Error
+
+from reservations.models import (
+    STATE_CHOICES,
+    RecurringReservation,
+    Reservation,
+    ReservationPurpose,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def create_reservations_from_allocation_results(application_event):
+    for schedule in application_event.application_event_schedules.all():
+        create_reservation_from_schedule_result(
+            schedule.applicationeventscheduleresult, application_event
+        )
+
+
+def create_reservation_from_schedule_result(result, application_event):
+    recurring_reservation = RecurringReservation.objects.create(
+        user=application_event.application.user,
+        application=application_event.application,
+        age_group=result.application_event_schedule.application_event.age_group,
+        ability_group=result.application_event_schedule.application_event.ability_group,
+    )
+
+    reservation_date = application_event.begin
```

Comment:
I think there is a bug here, running this with 

```
day = 0
reservation_date = datetime.date(2021, 4, 27)
```

returns 2021-04-28 when it should return 2021-05-03 which is the next monday after 2021-04-27

I think something like this should be correct

```
def next_matching_weekday(date, weekday):
    days_ahead = weekday - date.weekday()
    if days_ahead < 0: 
        days_ahead += 7
    return date + datetime.timedelta(days_ahead)

print(next_matching_weekday(datetime.date(2021, 4, 27), 0))
```

Reply:
actually there is already basically this in 

date_util.py#next_or_current_matching_weekday

Reply:
very nicely catched :fishing_pole_and_fish:

Reply:
this should be now fixed.